plugin/includes/class-attachmentav-loader.php

```diff
@@ -209,19 +205,98 @@ function scan_attachment($post_id) {
 					if (array_key_exists('finding', $body)) {
 						add_post_meta($post_id, 'attachmentav_scan_finding', $body['finding']);
 					}
+				} else if ($response['response']['code'] == 401) {
+					add_post_meta($post_id, 'attachmentav_scan_result', 'error:license_key_missing');
+				} else if ($response['response']['code'] == 429) {
+					add_post_meta($post_id, 'attachmentav_scan_result', 'error:max_scans_reached');
 				} else {
-					if ($response['response']['code'] == 401) {
-						add_post_meta($post_id, 'attachmentav_scan_result', 'error:license_key_missing');
-					} else if ($response['response']['code'] == 429) {
-						add_post_meta($post_id, 'attachmentav_scan_result', 'error:max_scans_reached');
-					} else {
-						add_post_meta($post_id, 'attachmentav_scan_result', 'error:server');
-					}
+					add_post_meta($post_id, 'attachmentav_scan_result', 'error:server');
 				}
 			}
 			add_post_meta($post_id, 'attachmentav_scan_result', 'CLEAN');
 		}
-		add_action("add_attachment", 'scan_attachment');
+		add_action('add_attachment', 'scan_attachment');
+
+		function wpforms_scan_file($file) {
+			$endpoint = 'https://eu.developer.attachmentav.com/v1/scan/sync/download';
+			$args = array(
+				'timeout' => 30,
+				'body'    => '{"download_url":"' . $file['value'] . '"}',
+				'headers' => array(
+					'x-api-key' => get_option('attachmentav_api_key'),
+					'x-wordpress-site-url' => get_site_url(),
+					'Content-Type' => 'application/json',
+				),
+			);
+			$response = wp_remote_post( $endpoint, $args);
+			if (is_wp_error($response)) {
+				return array('status' => 'error:client');
+			} else {
+				if ($response['response']['code'] == 200) {
+					return json_decode($response['body'], true);
+				} else if ($response['response']['code'] == 401) {
+					return array('status' => 'error:license_key_missing');
+				} else if ($response['response']['code'] == 429) {
+					return array('status' => 'error:max_scans_reached');
+				} else {
+					return array('status' => 'error:server');
+				}
+			}
+		}
+		function wpforms_status2error($filename, $ret) {
+			if ($ret['status'] == 'infected') {
+				if (!empty($ret['finding'])) {
+					return 'The file ' . $filename . ' is infected and therefore blocked (' . $ret['finding'] . ')';
+				} else {
+					return 'The file ' . $filename . ' is infected and therefore blocked';
+				}
+			} else if ($ret['status'] == 'no') {
```

Comment:
There is a config option to define whether unscannable files should be accepted or blocked.